openLRSng/multiproto.h

```diff
@@ -0,0 +1,177 @@
+#ifndef _MULTI_H
+#define _MULTI_H_
+
+#define MULTI_HEADER 0x55
+#define MULTI_PROTOCOL_OPENLRS 0x1B
+#define MULTI_OPERATION_TIMEOUT_MS 5000
+
+#define MULTI_TELEMETRY_FRSKY_HUB    0x03
+#define MULTI_FLAGS_INPUT_DETECTED   0x01
+#define MULTI_FLAGS_SERIAL_ENABLED   0X02
+#define MULTI_FLAGS_PROTOCOL_VALID   0x04
+#define MULTI_FLAGS_BINDING          0x08
+#define MULTI_FLAGS_WAITING_FOR_BIND 0x08
+
+extern uint8_t serialFrameIndex;
+extern ppm_msg ppmWork;
+extern volatile uint8_t ppmAge;
+
+#ifdef DEBUG_DUMP_PPM
+extern uint8_t ppmDump;
+#endif
+
+uint8_t multiWork[3];
+bool multiBind = false;
+bool multiAutoBind = false;
+bool multiRangeCheck = false;
+bool multiLowPower = false;
+uint8_t multiProfile = 0;
+int8_t multiOption = 0;
+uint32_t multiLastProfileChange = 0;
+
+// prototypes
+bool newMultiProfileSelected(bool useTimeout);
+uint16_t MULTItoPPM(uint16_t input);
+void processMULTI(uint8_t c);
+void checkMultiState(void);
+uint8_t checkMultiPower(uint8_t power);
+uint8_t multiBindActivated(void);
+uint8_t getMultiProfile(void);
+void sendMultiTelemetry(bool inputDetected, bool protocolValid, bool isBinding);
+
+void sendMultiTelemetry(bool inputDetected, bool protocolValid, bool isBinding)
+{
+  rcSerial->write('M');
+  rcSerial->write('P');
+  rcSerial->write(MULTI_TELEMETRY_FRSKY_HUB);
+  rcSerial->write(5);
+
+  uint8_t flags = MULTI_FLAGS_SERIAL_ENABLED;
+  if (inputDetected) {
+    flags = flags | MULTI_FLAGS_INPUT_DETECTED;
+  }
+  if (protocolValid) {
+    flags = flags | MULTI_FLAGS_PROTOCOL_VALID;
+  }
+  if (isBinding) {
+    flags = flags | MULTI_FLAGS_BINDING;
+  }
+  rcSerial->write(flags);
+  rcSerial->write(version >> 8);
+  rcSerial->write((version >> 4) & 0x0f);
+  rcSerial->write(version & 0x0f);
+  rcSerial->write(0);
+}
+
+uint8_t checkMultiPower(uint8_t power)
+{
+  uint8_t p = power;
+
+  if (multiOption >= 0) {
+    p = multiOption & 0x07;
+  }
+
+  if (!multiLowPower) {
+    p = 7;
+  }
+
+  if (multiRangeCheck) {
+    if (p > 3) {
+      p -= 3;
+    } else {
+      p= 0;
+    }
+  }
+  return p;
+}
+
+bool newMultiProfileSelected(bool useTimeout)
+{
+  return ((multiProfile > 0) &&
+          (multiProfile <= TX_PROFILE_COUNT) &&
+          ((multiProfile - 1) != activeProfile) &&
+          (!useTimeout ||
+           ((millis() - multiLastProfileChange) >= MULTI_OPERATION_TIMEOUT_MS)));
```

Comment:
Indentation.

Reply:
ok, will fix

Reply:
actually, this is one statement where the lack of brackets somewhat confused things. I wasn't sure if the logic tests were being executed correctly; is the logical OR supposed to be at the same level as the logical AND's, or is it meant to be enclosed in the final AND condition? please let me know how this should work.

Reply:
lol, actually, I just realized that astyle is at fault for the weird indentation. It did it automatically, and over-wrote my changes. Let me know what you'd like to do here...

Reply:
Again, way too many brackets for my liking. :-)

But apart from that the indentation just looks messy as it is, and if it's generated by a tool we should look into setting the tool up to generate indentation with the same step width for every step.

Reply:
will fix...